### ISSUE TITLE
Help with Android backend receipt Validation.

### ISSUE THREAD

**Issue by matamicen** [dooboolab/react-native-iap#815]
Hi guys,

I couldn't find any nice and clear tutorial or example of how to do it? Is not clear if I have to create service account or an oauth2, and then How to call https://www.googleapis.com/auth/androidpublisher ? What parameters to pass, in the body? in the header? I just want to test it first from POSTman. 
 
Any help will be appreciated.

Thanks!

**Comment by matamicen** [comment]
Update:

We have finally called the Subscription API:

https://www.googleapis.com/androidpublisher/v2/applications/packageName/purchases/subscriptions/subscriptionId/tokens/token?access_token=you_access_token


But the api response always:

```
{
   "error": {
       "errors": [
           {
               "domain": "androidpublisher",
               "reason": "purchaseTokenDoesNotMatchPackageName",
               "message": "The purchase token does not match the package name.",
               "locationType": "parameter",
               "location": "token"
           }
       ],
       "code": 400,
       "message": "The purchase token does not match the package name."
   }
}
```

We passed the PurchaseToken and PackageName received from the Listener. 

@hyochan  Any ideas :) ?

Thanks!

**Comment by hyochan** [comment]
I'd like to share you wonderful [blog post](http://blog.naver.com/PostView.nhn?blogId=gusrn1423&logNo=221352943668&categoryNo=50&parentCategoryNo=0&viewDate=&currentPage=1&postListTopCurrentPage=1&from=search&userTopListOpen=true&userTopListCount=5&userTopListManageOpen=false&userTopListCurrentPage=1) written in Korean. If you think it is hard for you to get the idea, I'll try to translate in the future.

**Comment by Stafox** [comment]
@matamicen You should use Google API Client SDK on your backend.
https://developers.google.com/api-client-library

**Comment by Bang9** [comment]
@matamicen I made a [repository](https://github.com/Bang9/iap-android-validation) for refer, check this one

**Comment by matamicen** [comment]
@Bang9 @hyochan Thanks both guys!

We solved it, the problem was that the the API reply that the token didn't match the receipt or something like that, the only thing we did to solve it was to create a new product (subscription) and start working both products inmediately. it seems like Google Dashboard need a refresh or something like that.

Thanks for your support and I hope it helps someone in the future.
